app/components/date-input.js

```diff
@@ -0,0 +1,94 @@
+import Component from '@glimmer/component';
+
+import { tracked } from '@glimmer/tracking';
+import { guidFor } from '@ember/object/internals';
+
+import { restartableTask, timeout } from 'ember-concurrency';
+import moment from 'moment';
+
+import { INPUT_DEBOUNCE } from 'frontend-lmb/utils/constants';
+
+export default class DateInputComponent extends Component {
+  elementId = `date-${guidFor(this)}`;
+
+  @tracked dateInputString;
+  @tracked errorMessage;
+
+  constructor() {
+    super(...arguments);
+
+    if (this.args.value && isValidDate(this.args.value)) {
+      this.dateInputString = moment(this.args.value).format('DD-MM-YYYY');
+      this.processDate(this.args.value);
+    }
+  }
+
+  onChange = restartableTask(async (event) => {
+    await timeout(INPUT_DEBOUNCE);
+
+    const inputValue = event.target?.value;
+    this.dateInputString = inputValue;
+
+    let [day, month, year] = inputValue.split('-');
+    const formatForDateConstructor = `${month}-${day}-${year}`;
+    const date = this.processDate(new Date(formatForDateConstructor) ?? null);
+
+    this.args.onChange?.(date);
+  });
+
+  isDateInRange(date, min, max) {
+    if (!date) {
+      return false;
+    }
+
+    date.setHours(0, 0, 0, 0);
```

Comment:
would rework this so it uses moment, this setHours thing got us in trouble before in another story

Reply:
Done!